core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/FcJsonRpc.java

```diff
@@ -107,34 +109,32 @@ public boolean filecoinWalletVerify(
       @JsonRpcParam("address") final String filecoinAddress,
       @JsonRpcParam("data") final String dataToVerify,
       @JsonRpcParam("signature") final FilecoinSignature filecoinSignature) {
-    final Optional<ArtifactSigner> signer = fcSigners.getSigner(filecoinAddress);
+    final FilecoinAddress address = FilecoinAddress.fromString(filecoinAddress);
+    final Bytes signature = Bytes.fromBase64String(filecoinSignature.getData());
 
-    if (signer.isPresent()) {
-      final Bytes bytesToVerify = Bytes.fromBase64String(dataToVerify);
-      final ArtifactSignature artifactSignature = createArtifactSignature(filecoinSignature);
-      return signer.get().verify(bytesToVerify, artifactSignature);
-    } else {
-      throw new FilecoinSignerNotFoundException();
+    switch (address.getProtocol()) {
+      case SECP256K1:
```

Comment:
do you need to verify that the address and the "signature type" align - i.e. both are BLS?